core/src/main/java/org/apache/druid/java/util/common/DateTimes.java

```diff
@@ -107,7 +107,23 @@ public static DateTime utc(long instant)
 
   public static DateTime of(String instant)
   {
-    return new DateTime(instant, ISOChronology.getInstanceUTC());
+    try {
+      Integer seconds = Integer.valueOf(instant);
```

Comment:
This code won't work for timestamps that are near the epoch. I think it's a bit too clever and we should stick to only working on milliseconds timestamps and ISO8601 strings.

Reply:
As I wrote it last night I realized it's doing too much.

I'll work on it today.